DTML.EduBot/Controllers/MessagesController.cs

```diff
@@ -21,19 +21,27 @@ public class MessagesController : ApiController
         /// </summary>
         public async Task<HttpResponseMessage> Post([FromBody]Activity activity)
         {
-            if (activity.Type == ActivityTypes.Message)
+            try
             {
-                using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+                if (activity.Type == ActivityTypes.Message)
                 {
-                    await Conversation.SendAsync(activity, () => scope.Resolve<RootDialog>());
+                    using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+                    {
+                        await Conversation.SendAsync(activity, () => scope.Resolve<RootDialog>());
+                    }
+                }
+                else
+                {
+                    await HandleSystemMessageAsync(activity);
                 }
+                var response = Request.CreateResponse(HttpStatusCode.OK);
+                return response;
             }
-            else
+            catch(Exception e)
```

Comment:
Is this for global exception handling? I'm confused to see we are logging user id and then  just re-throws...